src/components/Tabs/Tabs.tsx

```diff
@@ -65,66 +61,37 @@ export const Tabs = ({
   onChange,
   ...other
 }: Props) => {
-  const ref = useRef<number | undefined>();
+  const getUID = useUIDSeed();
+  const activeTabPanelId = useUID();
   const headerRef = useRef<HTMLDivElement>(null);
   const [activeIndexState, setActiveIndexState] = useState(activeIndex);
   const [scrollableLeft, setScrollableLeft] = useState<boolean>(false);
   const [scrollableRight, setScrollableRight] = useState<boolean>(false);
 
-  /** Get all children that are Tab components. Ignore any others. */
-  const getTabs = useCallback(() => {
+  /** Children that are actually tabs. Any others are ignored. */
+  const tabs = useMemo(() => {
     return allByType(children, Tab);
   }, [children]);
 
-  const tabRefs = getTabs().map(() => React.createRef<HTMLAnchorElement>());
-  const [tabPanelIds, setTabPanelIds] = useState<string[]>([]);
-  const [tabIds, setTabIds] = useState<string[]>([]);
-  const getUID = useUIDSeed();
+  const tabRefs = useMemo(
+    () => tabs.map(() => React.createRef<HTMLAnchorElement>()),
+    [tabs],
+  );
 
-  /**
-   * Get previous prop
-   *
-   * This is used to compare the previous prop to the current prop.
-   */
-  function usePrevious(activeIndex: number) {
-    useEffect(() => {
-      ref.current = activeIndex;
-    });
-    return ref.current;
-  }
+  const tabIds = useMemo(
+    () => tabs.map((tab) => tab.props['aria-labelledby'] || getUID(tab)),
+    [tabs, getUID],
+  );
 
-  /**
-   * Set prevActiveIndex to previous value prop
-   *
-   * If prevActiveIndex is defined and previous value prop is not equal
-   * to current value prop, toggle state.
-   */
+  // Set the active tab if the `activeIndex` prop changes.
   const prevActiveIndex = usePrevious(activeIndex);
   useEffect(() => {
-    if (
-      (prevActiveIndex !== undefined || null) &&
-      prevActiveIndex !== activeIndex
-    ) {
+    if (prevActiveIndex != null && prevActiveIndex !== activeIndex) {
       setActiveIndexState(activeIndex);
       tabRefs[activeIndex].current?.focus();
     }
   }, [prevActiveIndex, activeIndex, tabRefs]);
```

Comment:
I'm tempted to delete `usePrevious` and just do something like

```ts
useEffect(() => {
  if (activeIndex != null) {
    setActiveIndexState(activeIndex);
    tabRefs[activeIndex].current?.focus();
  }
}, [activeIndex, tabRefs]);
```

which would do the correct thing when the `activeIndex` prop changes.

However, I think that would get messed up if tabs are dynamically added or removed (by activating the original `activeIndex` tab, instead of the currently active one).